ui.js

```diff
@@ -78,80 +63,59 @@ const Search = ({query, emojis, skinNumber, selectedIndex, onChangeQuery}) => {
 	);
 };
 
-class Emoj extends React.PureComponent {
-	constructor(props) {
-		super(props);
-		autoBindReact(this);
-
-		this.state = {
-			stage: STAGE_CHECKING,
-			query: '',
-			emojis: [],
-			skinNumber: props.skinNumber,
-			selectedIndex: 0,
-			selectedEmoji: null
-		};
-	}
-
-	render() {
-		const {
-			stage,
-			query,
-			emojis,
-			skinNumber,
-			selectedIndex,
-			selectedEmoji
-		} = this.state;
-
-		return (
-			<Box>
-				{stage === STAGE_COPIED && <CopiedMessage emoji={selectedEmoji}/>}
-				{stage === STAGE_SEARCH && (
-					<Search
-						query={query}
-						emojis={emojis}
-						skinNumber={skinNumber}
-						selectedIndex={selectedIndex}
-						onChangeQuery={this.handleChangeQuery}
-					/>
-				)}
-			</Box>
-		);
-	}
+const Emoj = ({skinNumber: initialSkinNumber, onSelectEmoji}) => {
+	const {exit} = useApp();
+	const [stage, setStage] = useState(STAGE_CHECKING);
+	const [query, setQuery] = useState('');
+	const [emojis, setEmojis] = useState([]);
+	const [skinNumber, setSkinNumber] = useState(initialSkinNumber);
+	const [selectedIndex, setSelectedIndex] = useState(0);
+	const [selectedEmoji, setSelectedEmoji] = useState(null);
+
+	useEffect(() => {
+		if (selectedEmoji && stage === STAGE_COPIED) {
+			onSelectEmoji(selectedEmoji);
+		}
+	}, [selectedEmoji, stage, onSelectEmoji]);
 
-	componentDidMount() {
-		this.setState({stage: STAGE_SEARCH}, () => {
-			this.props.stdin.on('data', this.handleInput);
-		});
-	}
+	const changeQuery = useCallback(query => {
+		setSelectedIndex(0);
+		setEmojis([]);
+		setQuery(query);
+	});
 
-	handleChangeQuery(query) {
-		this.setState({
-			query,
-			emojis: [],
-			selectedIndex: 0
-		});
+	useEffect(() => {
+		setStage(STAGE_SEARCH);
+	}, []);
+
+	useEffect(() => {
+		if (query.length <= 1) {
+			return;
+		}
+
+		debouncer(async () => {
```

Comment:
Any debouncer hook we could use? Might make this slightly nicer.

Alternatively, maybe `useInput` could have a `debounce` option? Seems like a common use-case to want to debounce input.

Reply:
> Any debouncer hook we could use? Might make this slightly nicer.

Good point.

> Alternatively, maybe useInput could have a debounce option? Seems like a common use-case to want to debounce input.

Could be worth adding, going to wait and see if this comes up again.